RecoEgamma/PhotonIdentification/plugins/PhotonMVANtuplizer.cc

```diff
@@ -108,27 +105,56 @@ class PhotonMVANtuplizer : public edm::one::EDAnalyzer<edm::one::SharedResources
       // config
       const bool isMC_;
       const double ptThreshold_;
+      const double deltaR_;
 
       // for AOD or MiniAOD case
       MultiTokenT<edm::View<reco::Photon>>        src_;
       MultiTokenT<std::vector<reco::Vertex>>      vertices_;
       MultiTokenT<std::vector<PileupSummaryInfo>> pileup_;
+      MultiTokenT<edm::View<reco::GenParticle>> genParticles_;
 
 };
 
-//
-// constants, enums and typedefs
-//
+enum PhotonMatchType {
+  FAKE_PHOTON,
+  TRUE_PROMPT_PHOTON,
+  TRUE_NON_PROMPT_PHOTON,
+};
 
-//
-// static data member definitions
-//
+namespace {
+
+    int matchToTruth( const reco::Photon& ph,
+                      const edm::View<reco::GenParticle>& genParticles,
+                      double deltaR)
+    {
+      // Find the closest status 1 gen photon to the reco photon
+      double dR = 999;
+      reco::GenParticle const * closestPhoton;
+      for (auto & particle : genParticles) {
+        // Drop everything that is not photon or not status 1
+        if( abs(particle.pdgId()) != 22 || particle.status() != 1 ) continue;
+
+        double dRtmp = ROOT::Math::VectorUtil::DeltaR( ph.p4(), particle.p4() );
+        if( dRtmp < dR ) {
+          dR = dRtmp;
+          closestPhoton = &particle;
+        }
+      }
+      // See if the closest photon (if it exists) is close enough.
+      // If not, no match found.
+      if(dR < deltaR) {
+          if( closestPhoton->isPromptFinalState() ) return TRUE_PROMPT_PHOTON;
```

Comment:
this is triggering a new issue in the static analyzer.
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25218/31629/llvm-analysis/report-1f898d.html#EndPath

Please fix.
A somewhat easy alternative is to move this code to /test subdirectory.